qiskit/chemistry/ground_state_calculation/adapt_vqe.py

```diff
@@ -0,0 +1,275 @@
+            vqe.initial_point = theta
+            raw_vqe_result = vqe.compute_minimum_eigenvalue(operator)
+            theta = raw_vqe_result.optimal_point.tolist()
+        else:
+            # reached maximum number of iterations
+            max_iterations_exceeded = True
+            logger.info("Maximum number of iterations reached. Finishing.")
+            logger.info("Final maximum gradient: %s", str(np.abs(max_grad[0])))
+
+        # once finished evaluate auxiliary operators if any
+        if aux_operators is not None and aux_operators:
+            vqe.compute_minimum_eigenvalue(operator, aux_operators)
+
+        if threshold_satisfied:
+            finishing_criterion = 'Threshold converged'
+        elif alternating_sequence:
+            finishing_criterion = 'Aborted due to cyclicity'
+        elif max_iterations_exceeded:
+            finishing_criterion = 'Maximum number of iterations reached'
+        else:
+            raise AquaError('The algorithm finished due to an unforeseen reason!')
+
+        # extend VQE returned information with additional outputs
+        result = AdaptVQEResult()
+        result.raw_result = raw_vqe_result
+        result.computed_electronic_energy = raw_vqe_result.eigenvalue.real
+        result.aux_values = raw_vqe_result.aux_operator_eigenvalues
+        result.num_iterations = iteration
+        result.final_max_gradient = max_grad[0]
+        result.finishing_criterion = finishing_criterion
+
+        self.transformation.add_context(result)
+
+        logger.info('The final energy is: %s', str(result.computed_electronic_energy))
+        return result
+
+
+class AdaptVQEResult(FermionicGroundStateResult):
+    """ AdaptVQE Result."""
+
+    @property
+    def num_iterations(self) -> int:
+        """ Returns number of iterations """
+        return self.get('num_iterations')
+
+    @num_iterations.setter
+    def num_iterations(self, value: int) -> None:
+        """ Sets number of iterations """
+        self.data['num_iterations'] = value
+
+    @property
+    def final_max_gradient(self) -> float:
+        """ Returns final maximum gradient """
+        return self.get('final_max_gradient')
+
+    @final_max_gradient.setter
+    def final_max_gradient(self, value: float) -> None:
+        """ Sets final maximum gradient """
+        self.data['final_max_gradient'] = value
+
+    @property
+    def finishing_criterion(self) -> str:
+        """ Returns finishing criterion """
+        return self.get('finishing_criterion')
+
+    @finishing_criterion.setter
+    def finishing_criterion(self, value: str) -> None:
+        """ Sets finishing criterion """
+        self.data['finishing_criterion'] = value
+
+    def __getitem__(self, key: object) -> object:
+        if key == 'final_max_grad':
+            warnings.warn('final_max_grad deprecated, use final_max_gradient property.',
+                          DeprecationWarning)
+            return super().__getitem__('final_max_gradient')
```

Comment:
We can probably remove this `DeprecationWarning` because this was only needed in the previous result class that extended the `VQEResult`.

Reply:
Resolved and about to push.